### DISCUSSION TITLE
Loading lists of nested components leads to client "component not found" errors

### DISCUSSION THREAD

**Discussion by jplhomer** [livewire/livewire#8069] [7. Bugs (archived)]
### Livewire version

v3.4.6

### Laravel version

v10.45.1

### Which PHP version are you using?

PHP 8.2

### Steps To Reproduce

1. Create a parent Livewire component which contains an array of items
2. In the view, loop through that list of items and pass their values to a child Livewire component
3. In the child Livewire component, render the value passed from the parent
4. Mimic network latency by adding a `sleep` in the parent `render` method like ~0.5 or so
5. Implement a feature which randomizes or refreshes the list values (e.g. a live search or a shuffle button)
6. **Slowly** start typing or removing characters to attempt to reproduce the behavior

### Repository or Wirebox URL

https://wirebox.app/b/4q2e6

### Additional information

Here's a video:

https://github.com/livewire/livewire/assets/848147/8b28b2e1-09ab-4b4f-980e-6a1da8f58523

It's frustrating that it's not a simple reproduction case. My gut says it's related to the JS code, likely a race condition which is responsible for hydrating the elements when they come back from the server competing with the code destroying the components for the next version from the server.

I've also tinkered with request isolation for the lazy children as well as using `:key` instead of `wire:key`. None seem to make a difference.

### Please confirm (incomplete submissions will not be addressed)

- [X] I have provided easy and step-by-step instructions to reproduce the bug.
- [X] I have provided code samples as text and NOT images.
- [X] I understand my bug report will be removed if I haven't met the criteria above.

**Comment by EPuigdemont** [comment]
I am having the same issue, I load data from an external API which is a list, each item in the list is its own Livewire component that is built from the passed props in the mount method. When one of the items in the list is removed from the API response, when reloading the last in the parent component it "causes the component not found" error.
So far I have made the :key in the foreach loop to be a hash string generated from the api data, so whenever there is a change the data the hash changes and it forces a re-render of the child, but this is not ideal.

**Comment by liaosankai** [comment]
I previously encountered this issue, and after extensive testing, I found that when using the nesting approach to load child components, there must not be any DOM elements wrapping the child component; otherwise, an "uncaught component not found" error will occur during re-rendering.

Wrong:
```
<!-- child -->
<div>
    {{ $word }}
</div>
```
```blade
<ul>
@foreach ($things as $item)
    <li>
        <livewire:child :word="$item->word" :key="'item-' . $item->id" />
    </li>
@endforeach
</ul>
```

Correct:
```
<!-- child -->
<li><div>{{ $word }}</div></li>
```
```blade
<ul>
@foreach ($things as $item)
    <livewire:child :word="$item->word" :key="'item-' . $item->id" />
@endforeach
</ul>
```

@jplhomer 

While working with nested components, I noticed that the documentation does not mention a potential issue: when using the nesting approach to load child components, there should not be any DOM elements wrapping the child component. If there are, it can cause unexpected rendering issues.

Would it be possible to add a note or reminder about this point in the relevant section of the documentation? I believe this would help other developers avoid confusion and save time during implementation.

**Reply by joshhanley** [reply]
@liaosankai actually the issue is that you don't have any `wire:key` on the root element in the loop.

If you do this, it should work

```blade
<ul>
@foreach ($things as $item)
    <li wire:key="item-{{ $item->id }}">
        <livewire:child :word="$item->word" :key="'child-item-' . $item->id" />
    </li>
@endforeach
</ul>
```

**Comment by joshhanley** [comment]
We've recently done a bunch of work to try and fix scenarios that cause this error. This will be available in the next release.
